scompose/client/check.py

```diff
@@ -25,10 +25,10 @@ def main(args, parser, extra):
 
     # validate compose files
     for f in args.file:
-        result = validate_config(f)
-        if not result and not args.preview:
+        valid = validate_config(f)
+        if valid and not args.preview:
             bot.info("%s is valid." % f)
-        elif result:
```

Comment:
Same question as before! What about the case when it's valid and a preview and the error says mistakenly "X is not valid?"

Reply:
Oops 😅 good catch. On further reflection, there is no reason why args.preview should be part of that if stmt, right?

Reply:
In my head, preview is only about whether we should print the deep-merged config file. 

If the attempt of using it here was to make it behave like a "quiet mode" check (for scripting purposes) then the previous impl needs a different tweak as the exception thrown would not let that happen as it is, right?

Reply:
Okay, I took a look at the earlier PRs and the intent of having args.preview was to have the "quiet mode"... I will add the `elif result` back. That should address that point

Reply:
I updated the logic now. My brain stops working after 22ish..so if I missed anything, let me know :)

Reply:
haha no worries! I'm at 2am over here I really need to stop programming... :laughing: ... can't... !

Reply:
Wow! I hope I'm not the one disturbing your sleeping pattern... this PR can wait for sure 😅